uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -141,16 +135,18 @@ define([
       vm.initChildController()
       .then(function () {
         return $q.all([
-          vm._loadCalendar(),
+          loadCalendar(),
           loadDayTypes()
         ]);
       })
       .then(initDates)
```

Comment:
this should happen explicitly when the request is not being created, so you can simplify the function
```js
.then(function () {
  if (!vm.isMode('create')) {
    return initTimes();
  }
})

// or..
.then(!vm.isMode('create') && initDates)
```
```js
function initDates () {
  var attributes;

  attributes = vm.request.attributes();
  vm.uiOptions.fromDate = convertDateFormatFromServer(vm.request.from_date);

  return loadDayTypesForDate(vm.uiOptions.fromDate, 'from')
    .then(function () {
      // ...
    });
}
```

Reply:
Awesome idea. Done.